presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -336,4 +435,9 @@ public static int findAggregationStateParamId(Method method, int id)
         }
         return methods.build();
     }
+
+    static boolean isAggregationMetaAnnotation(Annotation annotation)
```

Comment:
Grouping `BlockIndex`, `AggregationState` and the meta parameters into one category, and calling the category `AggregationMetaAnnotation` is not an intuitive abstraction. In contrast, It is obvious what `isMetaParameter` would evaluate true for.

It looks like you are looking for anything that is not `@SqlType` here.

What about doing just that instead (so that you don't need this terminology and function)?